aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -135,6 +162,29 @@ TORCH_META_FUNC(fmin) (const Tensor& self, const Tensor& other) {
     build_binary_op(maybe_get_output(), self, other);
 }
 
+void comparison_op_check(const Tensor& self, const Tensor& other) {
+  // Validate that is possible to convert zero-dim tensor's dtype to other dtype
+  // without overflow
+  if (self.scalar_type() != other.scalar_type()) {
+    if (self.dim() != 0 && other.dim() == 0) {
+      native::check_convert(other.item(), self.scalar_type());
+    } else if (self.dim() == 0 && other.dim() != 0) {
+      native::check_convert(self.item(), other.scalar_type());
+    }
+  }
+}
+
+TORCH_META_FUNC2(eq, Tensor)(const Tensor& self, const Tensor& other) {
+  comparison_op_check(self, other);
+  build_comparison_op(maybe_get_output(), self, other);
+}
+
+TORCH_META_FUNC2(eq, Scalar)(const Tensor& self, const Scalar& other) {
+  auto other_tensor = native::wrapped_scalar_tensor_and_check_convert(other, self);
+  comparison_op_check(self, other_tensor);
```

Comment:
with wrapped_scalar_tensor_and_check_convert do you really need comparison_op_check?

Reply:
huh - if `self` is Scalar, won't that mean that both `self` and `other` have `dim() == 0`, which will make `comparison_op_check` a noop? Since it only does the check if exactly one of the two arguments has zero dimension.

Reply:
that might be true

Reply:
That makes sense. I will remove the call